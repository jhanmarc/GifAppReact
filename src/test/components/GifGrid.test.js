
import React  from 'react';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import "@testing-library/jest-dom"



import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');


describe("Pruebas en <GifGrid>", () => {

    const category = 'Dragon Ball';

    test('debe de mostrar el componente correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })
        
        const wrapper = shallow( <GifGrid category={category} /> )
        expect( wrapper ).toMatchSnapshot();
    });

    test('debe de mostrar items cuando se carga imagenes useFetchGifs', () => {
        
        const gifs = [
            { id: 'ABC',
              url: 'https://localhost/goku.jpg',
              title: 'Goku'
            },
            { id: '1234',
              url: 'https://localhost/goku.jpg',
              title: 'Goku'
            }
        ]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })
        
        const wrapper = shallow( <GifGrid category={category} /> )

        // expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length )

    })
    

})