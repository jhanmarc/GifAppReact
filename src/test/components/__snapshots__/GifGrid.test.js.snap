// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid> debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3>
    Dragon Ball
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
